utils/src/main/java/software/amazon/awssdk/utils/http/SdkHttpUtils.java

```diff
@@ -431,31 +431,23 @@ public static List<String> splitQueryString(String queryString) {
      */
     public static Set<String> parseNonProxyHostsProperty() {
         String systemNonProxyHosts = ProxySystemSetting.NON_PROXY_HOSTS.getStringValue().orElse(null);
-        return extractNonProxyHostsFromSystemProperty(systemNonProxyHosts);
+        return extractNonProxyHosts(systemNonProxyHosts);
     }
 
-    private static Set<String> extractNonProxyHostsFromSystemProperty(String systemNonProxyHosts) {
-        if (systemNonProxyHosts != null && !isEmpty(systemNonProxyHosts)) {
-            return Arrays.stream(systemNonProxyHosts.split("\\|"))
+    private static Set<String> extractNonProxyHosts(String nonProxyHosts) {
+        if (nonProxyHosts != null && !isEmpty(nonProxyHosts)) {
+            return Arrays.stream(nonProxyHosts.split("\\|"))
                          .map(String::toLowerCase)
                          .map(s -> StringUtils.replace(s, "*", ".*?"))
                          .collect(Collectors.toSet());
         }
         return Collections.emptySet();
     }
 
-    private static Set<String> extractNonProxyHostsFromEnvironmentVariable(String environmentNonProxyHosts) {
-        if (environmentNonProxyHosts != null && !isEmpty(environmentNonProxyHosts)) {
-            return Arrays.stream(environmentNonProxyHosts.split(","))
-                         .map(String::toLowerCase)
-                         .map(String::trim)
-                         .collect(Collectors.toSet());
-        }
-        return Collections.emptySet();
-    }
-
     public static Set<String> parseNonProxyHostsEnvironmentVariable() {
-        String environmentNonProxyHosts = ProxyEnvironmentSetting.NO_PROXY.getStringValue().orElse(null);
-        return extractNonProxyHostsFromEnvironmentVariable(environmentNonProxyHosts);
+        String hosts = ProxyEnvironmentSetting.NO_PROXY.getStringValue()
+                                                       .map(noProxyHost -> noProxyHost.replace(",", "|"))
```

Comment:
After offline Discussion with @zoewangg  updated this to be consistent as [V1](https://github.com/aws/aws-sdk-java/blob/a186eadd5376908b472993353dfc1b115e61dc1b/aws-java-sdk-core/src/main/java/com/amazonaws/http/apache/SdkProxyRoutePlanner.java#L47-L55) supporting both commas and pipes.